docs/src/references/mappers/auto_registration.md

```diff
@@ -0,0 +1,31 @@
+# Auto Registration of an entity
+
+Before any data messages from an entity can be processed, the entity has to be registered first.
+The entity can be registered either explicitly or implicitly (Auto registration).
+
+In `auto-registration`, the registration does not need a separate registration message, but the registration message will be derived from the first message that is sent by that entity.
+
+For example, sending a measurement message to te/device/child1///m/temperature, will result in the auto-registration of the entity with topic id: device/child1// and the auto-generated external id: <main-device-id>:device:child1, derived from the topic id. Similarly, a measurement message on te/device/child1/service/my-service/m/temperature, results in the auto-registration of both the device entity: device/child1// and the service entity: device/child1/service/my-service with their respective auto-generated external IDs, in that order.
```

Comment:
```suggestion
For example, sending a measurement message to `te/device/child1///m/temperature` will result in the auto-registration of the device entity with topic id: `device/child1//` and the auto-generated external id: `<main-device-id>:device:child1`, derived from the topic id.
Similarly, a measurement message on `te/device/child1/service/my-service/m/temperature` results in the auto-registration of both
the device entity: `device/child1//` and the service entity: `device/child1/service/my-service` with their respective auto-generated external IDs, in that order.
```
Splitting single-line paragraphs and long sentences + highlighting the topics and IDs.